StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/OrderingRules/UsingCodeFixProvider.cs

```diff
@@ -531,6 +560,82 @@ private List<UsingDirectiveSyntax> GenerateUsings(List<UsingDirectiveSyntax> usi
                 {
                     var currentUsing = usingsList[i];
 
+                    if (qualifyNames)
+                    {
+                        currentUsing = currentUsing.ReplaceNodes(
+                            new[] { currentUsing.Name },
+                            (originalNode, rewrittenNode) =>
+                            {
+                                switch (originalNode.Kind())
+                                {
+                                case SyntaxKind.QualifiedName:
+                                case SyntaxKind.IdentifierName:
+                                case SyntaxKind.GenericName:
+                                    if (originalNode.Parent.IsKind(SyntaxKind.UsingDirective)
+                                        || originalNode.Parent.IsKind(SyntaxKind.TypeArgumentList))
+                                    {
+                                        var symbol = this.semanticModel.GetSymbolInfo(originalNode, cancellationToken: CancellationToken.None).Symbol;
+                                        if (symbol == null)
+                                        {
+                                            return rewrittenNode;
+                                        }
+
+                                        if (symbol is INamespaceSymbol)
+                                        {
+                                            // TODO: Preserve inner trivia
+                                            string fullName = symbol.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+                                            NameSyntax replacement = SyntaxFactory.ParseName(fullName);
+                                            if (!originalNode.DescendantNodesAndSelf().OfType<AliasQualifiedNameSyntax>().Any())
+                                            {
+                                                replacement = replacement.ReplaceNodes(
+                                                    replacement.DescendantNodesAndSelf().OfType<AliasQualifiedNameSyntax>(),
+                                                    (originalNode2, rewrittenNode2) => rewrittenNode2.Name);
+                                            }
```

Comment:
`SymbolDisplayFormat.FullyQualifiedFormat` expands things like `System` to `global::System`. This code removes the `global::` qualifier if the original code did not contain any aliases. Its purpose is to ensure that the following two cases do not lose their aliases during the qualification step:

```csharp
// Want to keep the alias:
using global::System;
// Want to keep the alias:
using other::System;
// Want to avoid adding an alias:
using System;
```

The code does result in one sub-optimal change, but I believe it is very rare:

```csharp
// This:
using MyFunc = global::System.Func<System.IntPtr>;
using MyAction = System.Action<global::System.IntPtr>;
// Becomes this:
using MyFunc = global::System.Func<global::System.IntPtr>;
using MyAction = global::System.Action<global::System.IntPtr>;
```

In other words, if *any* part of the input name had an alias qualifier, then *all* names in the result of the code fix will contain alias qualifiers.